optuna/_optimize.py

```diff
@@ -209,29 +212,75 @@ def _run_trial(
             return trial
         raise
 
+    checked_values, failure_message = _check_and_convert_to_values(
+        len(study.directions), value_or_values, trial
+    )
+
+    if failure_message is None:
+        assert checked_values is not None
+        study._storage.set_trial_values(trial_id, checked_values)
+        study._storage.set_trial_state(trial_id, TrialState.COMPLETE)
+        study._log_completed_trial(trial, checked_values)
+    else:
+        study._storage.set_trial_system_attr(trial_id, "fail_reason", failure_message)
+        study._storage.set_trial_state(trial_id, TrialState.FAIL)
+        _logger.warning(failure_message)
+
+    return trial
+
+
+def _check_and_convert_to_values(
+    n_objectives: int, original_value: Union[float, Sequence[float]], trial: trial_module.Trial
+) -> Tuple[Optional[Sequence[float]], Optional[str]]:
+    if isinstance(original_value, Sequence):
+        if n_objectives != len(original_value):
+            return None, (
+                "Trial {} failed, because the number of the values {} is did not match the "
+                "number of the objectives {}.".format(
+                    trial.number, len(original_value), n_objectives
+                )
```

Comment:
Not a strong opinion, but I think we can simplify this message by using f-string.